### PR TITLE
feat(onyx-737): swipe to delete alert

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-737]

### Description

Swipe to delete alerts:

<video src="https://github.com/artsy/eigen/assets/3934579/82e0e9e5-81f4-43c6-bbff-a45df9e38a28" width="400px" />

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
  - [ ] I don't have android - could someone please check on it? 
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Swipe-to-delete feature on alerts list screen - @nickskalkin 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-737]: https://artsyproduct.atlassian.net/browse/ONYX-737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by nickskalkin** [comment]
@MounirDhahri I'm still fighting with tests by this PR is ready for pre-review! Huge props to you again for the knowledge share on animations

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Swipe-to-delete feature on alerts list screen - @nickskalkin)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5554852ef0b707166874626cb73f043ede2020fd
</p>

File: src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tests.tsx [hunk 1]
```diff
@@ -143,4 +145,38 @@ describe("SavedSearches", () => {
...
+          },
+        }),
+      })
+
+      fireGestureHandler<PanGesture>(getByGestureTestId(`pan-alert-banksy`), [
+        { translationX: 0 },
+        { translationX: -100 },
+      ])
+
+      expect(screen.getByTestId(`delete-button-banksy`)).toBeVisible()
```
Comment (nickskalkin) [inline src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tests.tsx:171]: to be fair - this is always "visible", even without swiping. This button has "absolute" position and hidden behind the alert item before swipe, but `.toBeVisible()` doesn't understand this.

File: src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchListItem.tsx [hunk 2]
```diff
@@ -1,34 +1,175 @@
...
+        : event.translationX
+    })
+    .onEnd(() => {
+      // If the user swipes more than half of the delete button width, show the delete button fully
+      if (translateX.value < -DELETE_BUTTON_WIDTH / 2) {
+        translateX.value = withTiming(-DELETE_BUTTON_WIDTH)
+        isDeleteButtonVisible.value = true
+      } else {
+        // Otherwise, hide the delete button
+        translateX.value = withTiming(0)
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchListItem.tsx:81]: nice idea

**Review by MounirDhahri — APPROVED** [review]
Really nice job!

**Review by dariakoko — APPROVED** [review]
Great job!

File: src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchListItem.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nickskalkin) [reply src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchListItem.tsx]: added a light ease @damassi :

https://github.com/artsy/eigen/assets/3934579/2ecdbd02-9a32-4de8-a48b-d5a50452fa51

[the remark this answers is not in the document]
Comment (damassi) [reply src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchListItem.tsx]: Much better! thanks for the update 👍

**Pull request merged**